lib/feedback_api_web/facades/survey_create_facade.ex

```diff
@@ -0,0 +1,65 @@
+defmodule FeedbackApiWeb.SurveyCreateFacade do
+  alias FeedbackApi.{Survey, User, Question, Group, Repo}
+  import Ecto.Query
+
+  def create_survey(arguments) do
+    Repo.transaction(fn ->
+      try do
+        survey = Survey.changeset(%Survey{}, arguments) |> Repo.insert!()
+        create_groups(survey, arguments["groups"])
+        create_questions(survey, arguments["questions"])
+      rescue
+        e -> Repo.rollback("Missing required fields")
+      end
+    end)
+  end
+
+  defp create_groups(survey, groups) do
+    Enum.map(groups, fn group -> create_group(survey, group) end)
+  end
+
+  defp create_questions(survey, questions) do
+    Enum.map(questions, fn question -> create_question(survey, question) end)
+  end
+
+  defp create_group(survey, group) do
+    new_group =
+      Group.changeset(%Group{}, group) |> Repo.insert!() |> Repo.preload([:survey, :users])
+
+    users = Repo.all(from u in User, where: u.id in ^group["members_ids"])
```

Comment:
Could you spell out line 29 for me?

Reply:
It's finding all records (u) in the Users table where the primary key is in the array of group members from the request.